scipy/stats/morestats.py

```diff
@@ -3478,3 +3478,72 @@ def circstd(samples, high=2*pi, low=0, axis=None, nan_policy='propagate'):
         R = np.minimum(1, hypot(sin_mean, cos_mean))
 
     return ((high - low)/2.0/pi) * sqrt(-2*log(R))
+
+
+def quartile_coeff_dispersion(samples, q=(0.25, 0.75), axis=0,
+                              interpolation='lower'):
+    """
+    Compute the quartile coefficient of dispersion for a given sample array.
+
+    Given two quartiles Q_low, Q_high from array samples such that
+    Q_low < Q_high, the quartile coefficient of dispersion is calculated by
+    (Q_high - Q_low)/(Q_high + Q_low).
+    by default it is computed using 3rd and 1st quartiles.
+
+    Parameters
+    ----------
+    samples : array_like
+        Input array.
+    q : array_like of float, optional
+        Sequence of 2 quantiles, such that q[0] < q[1].
+        The default is to compute with Q_high = 0.75, Q_low = 0.25.
+    axis : {int, tuple of int, None}, optional
+        Axis or axes along which the quantiles are computed.
+        The default is 0, computing quantiles along a flattened
+        version of the array.
+    interpolation : {'linear', 'lower', 'higher', 'midpoint', 'nearest'}
+        This optional parameter specifies the interpolation method to
+        use when the desired quantile lies between two data points
+        ``i < j``:
+            * linear: ``i + (j - i) * fraction``, where ``fraction``
+              is the fractional part of the index surrounded by ``i``
+              and ``j``.
+            * lower: ``i``.
+            * higher: ``j``.
+            * nearest: ``i`` or ``j``, whichever is nearest.
+            * midpoint: ``(i + j) / 2``.
+        The default is 'lower'.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Quartile_coefficient_of_dispersion
+
+    Returns
+    -------
+    dispersion : {float, array_like of float}
+        Dispersion for each axis slice.
+
+    Examples
+    --------
+    >>> from scipy.stats import quartile_coeff_dispersion
+    >>> quartile_coeff_dispersion(samples=[2, 4, 6, 8, 10, 12, 14, 16])
+    0.5
+
+    >>> quartile_coeff_dispersion(samples=[1.6, 2.1, 2.3, 2.4,
+    ...                           2.6, 2.9, 2.98, 3])
+    0.15999999999999998
+
+    """
+    if np.shape(q) != (2,):
+        raise ValueError("Shape of q must be (,2)")
+
+    if (q[0] > 1 or q[1] > 1) or (q[0] < 0 or q[1] < 0):
```

Comment:
```suggestion
    if q[0] < 0 or q[1] > 1:
```
I think the next condition should take care of the rest, right?

Reply:
Totally right.